### PR TITLE
#sql-tasks. Alexander Zenchyk

### PR DESCRIPTION
**Pull request by zeeenjaaa**
https://travis-ci.org/github/zeeenjaaa/db-assignments/builds/722285117

### CODE REVIEW

File: task/sql-tasks.js [hunk 1]
```diff
@@ -177,7 +273,13 @@ async function task_1_12(db) {
  *
  */
 async function task_1_13(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
```
Comment (AlexMokin) [inline task/sql-tasks.js:276]: that's good, but try to re-write query like this 

it's better to write this query like this

```
select
(select count()...) as 'T'
(select count()...) as 'TT'
```

notice that from clause is optional

File: task/sql-tasks.js [hunk 2]
```diff
@@ -280,7 +465,27 @@ async function task_1_21(db) {
...
+        INNER JOIN Products AS prod ON prod.ProductID = ordet.ProductID
+        WHERE ordet.UnitPrice = (
+            SELECT
+                MAX(ordet1.UnitPrice)
+            FROM Customers AS cust1
+            INNER JOIN Orders AS ord1 ON ord1.CustomerID = cust1.CustomerID
+            INNER JOIN OrderDetails AS ordet1 ON ordet1.OrderID = ord1.OrderID
+            WHERE cust.CustomerID = cust1.CustomerID 
+        )
+        GROUP BY PricePerItem, CompanyName, ProductName
```
Comment (AlexMokin) [inline task/sql-tasks.js]: just use distinct, you don't need group by if you don't aggregate anything

File: task/sql-tasks.js [hunk 3]
```diff
@@ -280,7 +465,27 @@ async function task_1_21(db) {
...
+        SELECT
+            cust.CompanyName AS 'CompanyName',
+            prod.ProductName AS 'ProductName',
+            ordet.UnitPrice AS 'PricePerItem'
+        FROM Customers AS cust
+        INNER JOIN Orders AS ord ON ord.CustomerID = cust.CustomerID
+        INNER JOIN OrderDetails AS ordet ON ordet.OrderID = ord.OrderID
+        INNER JOIN Products AS prod ON prod.ProductID = ordet.ProductID
+        WHERE ordet.UnitPrice = (
+            SELECT
```
Comment (AlexMokin) [inline task/sql-tasks.js]: try to group by this subquery customer id and join this subquery on unitprice and customer id

**Review by AlexMokin — COMMENTED** [review]
Looks good. See comments.
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by zeeenjaaa** [comment]
Thank you for review! I've got comments and made fixes.

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**